src/java/htsjdk/samtools/filter/InsertSizeFilter.java

```diff
@@ -0,0 +1,31 @@
+package htsjdk.samtools.filter;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * Filter things that fall outside a specified range of insert sizes.
+ * This will automatically omit unpaired reads.
+ */
+public class InsertSizeFilter implements SamRecordFilter {
+    final int minInsertSize;
+    final int maxInsertSize;
+
+    public InsertSizeFilter(final int minInsertSize, final int maxInsertSize) {
+        if (minInsertSize > maxInsertSize) throw new SAMException("Cannot have minInsertSize > maxInsertSize");
```

Comment:
The bounds are inclusive, so it'll include things with exactly that insert size and nothing else.